### ISSUE TITLE
SDL application fullscreen problem

### ISSUE THREAD

**Issue by i3bot** [i3/i3#1413]
[Originally reported by miguel.deval@…]
(When I start frozen-bubble 2.2.1beta1 in fullscreen mode in i3 4.8 (2014-06-15) that is available in Archlinux, the window is displayed in the original resolution (not fullscreen mode) with a black background. The GPU is an Intel graphics card.

**Comment by andaag** [comment]
I've seen similar things with some steam games, specifically valve's (HL2 engine). I for example had left 4 dead 2 configured to be fullscreen and it defaulted to fairly low resolution. When opening it it lagged and ... did not look low res. Checked my monitor, it was running at 2560x1440. Valve is pretty big, so I'm guessing this isn't an issue for their engine in general :p

**Comment by acrisci** [comment]
I can't reproduce this. Is the problem still in version 4.9?

**Comment by AndreasTPC** [comment]
I can reproduce this problem in i3 4.10.3 (the build that is in debian sid, debians package version is 4.10.3-1), not in frozen bubble but I ran into it while writing a program of my own that uses SDL.

I have reduced my code to a simple proof of concept which you can find here:
https://dl.dropboxusercontent.com/u/43723451/sdltest.tar.gz

To compile run it you just need SDL2. The code creates a 800x600 window and draws a red rectangle covering the entire window. When the window is resized SDL generates a SDL_WINDOWEVENT_RESIZED and the code then re-draws the red rectangle using the new image dimensions.

The problem is that when i3 does the initial resize of the window when it's created no SDL_WINDOWEVENT_RESIZED event is generated and SDL still thinks that the window has the initially specified dimensions. If you manually resize the window after the initial creation the event is however generated and the red square gets re-drawn at the proper size. When the same code is run on other tiling window managers the initial resize event does get generated properly.

Here are some screenshots illustrating the problem.
Initial window launch: https://dl.dropboxusercontent.com/u/43723451/sdltest-screen1.png
After resizing window manually: https://dl.dropboxusercontent.com/u/43723451/sdltest-screen2.png

Let me know if there's any other info you need to investigate this problem.

**Comment by stapelberg** [comment]
When running your example under xtrace with the default i3 config (note that config directives such as `workspace_layout stacked` lead to the issue not being triggered), I actually see the expected ConfigureNotify events being sent by i3 (unrelated log lines removed for readability):

```text
001:<:00b3:  8: Request(8): MapWindow window=0x02200008
001:>:00b3: Event ReparentNotify(21) event=0x02200008 window=0x02200008 parent=0x00e016e8 x=0 y=0 override-redirect=false(0x00)
001:>:00b3: Event ConfigureNotify(22) event=0x02200008 window=0x02200008 above-sibling=None(0x00000000) x=4 y=0 width=1912 height=2089 border-width=0 override-redirect=false(0x00)
001:>:00b3: Event MapNotify(19) event=0x02200008 window=0x02200008 override-redirect=false(0x00)
001:>:00b3: Event Expose(12) window=0x02200008 x=0 y=0 width=1912 height=2089 count=0x0000
001:<:00b4: 16: Request(12): ConfigureWindow window=0x02200008 values={stack-mode=Above(0x00)}
001:>:00b5: Event (generated) ConfigureNotify(22) event=0x02200008 window=0x02200008 above-sibling=None(0x00000000) x=1924 y=30 width=1912 height=2089 border-width=0 override-redirect=false(0x00)
001:>:00b5: Event (generated) ConfigureNotify(22) event=0x02200008 window=0x02200008 above-sibling=None(0x00000000) x=1924 y=30 width=1912 height=2089 border-width=0 override-redirect=false(0x00)
001:<:00b7:  8: Request(14): GetGeometry drawable=0x02200008
001:>:00b7:32: Reply to GetGeometry: depth=0x18 root=0x000002ea x=4 y=0 width=1912 height=2089 border-width=0
```

I tried looking into libsdl2 to see whether it’s a quick fix on their end, but when enabling debug messages, I see that they don’t even realize there are events up until a certain point.

To me, this looks like a race condition in libsdl2, where ConfigureNotify events are not handled properly for some reason.

I suggest you file a bug with libsdl2.

**Comment by AndreasTPC** [comment]
This is a bit embarassing, but apperantly I ran into this bug before, because a bug was already filed with libsdl over two year ago and I commented on it previously and forgot all about it. By coincidence I got an e-mail from their bugtracker about that bug being closed today. There's a one-line patch in their bug tracker that fixes the issue, but the sdl devs have not applied it, they closed the bug as "end of life" today since it was filed against an old version. I'm gonna see if I can get someone there to give it some attention. https://bugzilla.libsdl.org/show_bug.cgi?id=1859

I'm just posting this so people with the same issue can find the solution, since their bug is not easily found on google (I googled this issue a couple of days ago and didn't come across it).
